azure-iot-device/azure/iot/device/common/transport/mqtt/pipeline_stages_mqtt.py

```diff
@@ -58,7 +58,27 @@ def on_connected():
                 self.complete_op(op)
 
             self.provider.on_mqtt_connected = on_connected
-            self.provider.connect(self.sas_token)
+            try:
+                self.provider.connect(self.sas_token)
+            except BaseException:
+                self.provider.on_mqtt_connected = self.on_connected
```

Comment:
Also I'm just generally rather confused by what's going on this whole area. I'd appreciate an inline comment explaining.

Reply:
Not sure what you mean by "this whole area".  Hopefully my comment above explains it for you.  If not, please be more explicit about what's missing or confusing you.

---
In reply to: [287157524](https://github.com/Azure/azure-iot-sdk-python-preview/pull/94#discussion_r287157524) [](ancestors = 287157524)